### ISSUE TITLE
Pathfind

### ISSUE THREAD

**Issue by Dangleos** [Anuken/Mindustry-Suggestions#3015]
### Describe the content or mechanics you are proposing.

Pathfind should make units obey THEIR own default ai instead of dagger.
And add new version of pathfind that makes unit follow path to the specific tile with option to ignore walls for spider units or stay in water for marine

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

It would make processors controle ground units much better and allow to controle marine units. Right now marine units are very hard to control

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by OnyxFinger** [comment]
yes but how do you optimize it ?
a casual device simply isn't good enough to fluidly pathfind to a possibly constantly changing point

**Comment by Dangleos** [comment]
> yes but how do you optimize it ?
That is anuke's work

**Comment by OnyxFinger** [comment]
> > yes but how do you optimize it ?
> 
> That is anuke's work

yes, and it's not actually possible. I don't think making optimised code would fix the issue, and even if it does, it's probably not worth it

**Comment by hyx3179** [comment]
This is actually impossible. There is a research group in the School of Computer Science in our school that is studying automatic pathfinding, and they have published papers in journals.

**Comment by itcannotbe** [comment]
Send ~redacted~ links
